src/types/permissions.ts

```diff
@@ -32,50 +58,76 @@ export interface PermissionHandle {
   permissions: Permission[];
 }
 
-export class FungibleAssetPermission extends Serializable {
+type ToPayload<T> = { [K in keyof T]: string } & { readonly type: string };
+
+interface FungibleAssetPermissionProperties {
+  readonly asset: AccountAddress;
+  readonly amount: bigint | number;
+}
+type FungibleAssetPermissionPayload = ToPayload<FungibleAssetPermissionProperties>;
+
+export class FungibleAssetPermission extends MovePermission implements FungibleAssetPermissionProperties {
+  static readonly type = "FungibleAsset";
+
   readonly asset: AccountAddress;
 
   readonly amount: bigint;
 
-  constructor({ asset, amount }: { asset: AccountAddress; amount: number | bigint }) {
+  constructor({ asset, amount }: FungibleAssetPermissionProperties) {
     super();
     this.asset = asset;
     this.amount = BigInt(amount);
   }
 
-  static from(args: { asset: AccountAddress; amount: number | bigint }): FungibleAssetPermission {
-    return new FungibleAssetPermission(args);
+  static from(args: FungibleAssetPermissionProperties): FungibleAssetPermission;
+  static from(args: FungibleAssetPermissionPayload): FungibleAssetPermission;
+  static from(args: FungibleAssetPermissionPayload | FungibleAssetPermissionProperties): FungibleAssetPermission {
```

Comment:
Overloading `from` allows us to ingest the deserialized string/json object where every property is a `string` OR the proper types as if the dapp is creating a permission. Again, this is probably overcomplicated.